cms/search/signal_handlers.py

```diff
@@ -21,35 +23,101 @@ def get_publisher() -> KafkaPublisher | LogPublisher:
+    """Called after a Wagtail Page is moved in the tree.
+    "instance" is the moved Page object.
+    We use the publish_created_or_updated method to update search for the moved page and any of its non-excluded
+    descendants, which will also be affected by the move.
+    """
+    if kwargs["url_path_before"] == kwargs["url_path_after"]:
+        # No change in URL path, no need to update search index of the instance or descendants
+        return
+    if instance.get_view_restrictions().exists():
+        # Pages with view restrictions should not be exposed in search
+        # this is inherited by descendants, so nothing more to do
+        return
+    old_url_path: str = kwargs["url_path_before"]
+    new_url_path: str = kwargs["url_path_after"]
+    if instance.live and instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES:
+        try:
+            get_publisher().publish_created_or_updated(instance.specific_deferred, old_url_path=old_url_path)
+        except Exception:  # pylint: disable=broad-except
+            logger.exception(
+                "Error publishing moved page updated to search",
+                extra={"page_id": instance.id, "old_url_path": old_url_path, "new_url_path": new_url_path},
+            )
+
+    for moved_descendant in instance.get_descendants().filter(live=True):
+        if (
+            moved_descendant.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES
+            and not moved_descendant.get_view_restrictions().exists()
+        ):
+            old_descendant_path = build_old_descendant_path(
+                instance, moved_descendant, parent_path_before=old_url_path, parent_path_after=new_url_path
+            )
+
+            try:
+                get_publisher().publish_created_or_updated(
+                    moved_descendant.specific_deferred, old_url_path=old_descendant_path
+                )
+            except Exception:  # pylint: disable=broad-except
+                logger.exception(
+                    "Error publishing moved descendant page updated to search",
+                    extra={"page_id": moved_descendant.id, "old_url_path": old_url_path, "new_url_path": new_url_path},
+                )
+
+
+def build_old_descendant_path(
+    parent_page: "Page", descendant_page: "Page", parent_path_before: str, parent_path_after: str
+) -> str | None:
+    """Build the old URL path for a moved descendant page."""
+    if descendant_page.url_path.startswith(parent_path_after):
+        # We expect the old URL path to be derivable from the new URL path of the parent and the descendant
+        # Strip the url_path_after prefix from the descendant's url_path and prepend the url_path_before
+        return f"{parent_path_before}{descendant_page.url_path[len(parent_path_after) :]}"
+    logger.error(
+        "Found mismatching descendant page url_path while handling page move, cannot build old URL "
```

Comment:
Super nit: Can we add an extra newline above? I find these hard to read as they feel crammed.

Reply:
Addressed in [b8ba9d4](https://github.com/ONSdigital/dis-wagtail/pull/392/commits/b8ba9d40b700eb471fb18c6d06ce9f0a00df1dae).